e2e/src/test/scala/scalapb/zio_grpc/TestServiceSpec.scala

```diff
@@ -21,7 +23,15 @@ object TestServiceSpec extends DefaultRunnableSpec {
     ZLayer.fromServiceManaged { (ss: Server.Service) =>
       ZManaged.fromEffect(ss.port).orDie >>= { (port: Int) =>
         val ch = ZManagedChannel(
-          ManagedChannelBuilder.forAddress("localhost", port).usePlaintext()
+          ManagedChannelBuilder
+            .forAddress("localhost", port)
+            .defaultServiceConfig(
+              new Gson().fromJson(
```

Comment:
can the service be fully configured through code in here rather than read from JSON? Can you add a few comments describing how the test works and how the retry policy is being used.